### PR TITLE
doc: services: device_mgmt: mcumgr: Remove references to go tool

### PR DESCRIPTION
**Pull request by nordicjm**
The go tool documentation is very old, and the tool is not recommended for use, therefore remove all documentation relating to it to prevent people from using it

### CODE REVIEW

File: samples/subsys/mgmt/mcumgr/smp_svr/README.rst [hunk 1]
```diff
@@ -287,80 +237,47 @@ For clarity, here is DTS label to slot to ``<image>`` translation table:
...
-We can reset the device remotely to observe (use the console output) how
-MCUboot swaps the images:
-
-.. code-block:: console
-
-   sudo mcumgr <connection string> reset
-
-Upon reset MCUboot will swap slot-0 and slot-1.
+We can reset the device remotely to observe (use the console output) how MCUboot swaps the images,
+check the documentation in the tool of your choice. Upon reset MCUboot will swap slot-0 and
```
Comment (kartben) [inline samples/subsys/mgmt/mcumgr/smp_svr/README.rst:273]: non blocking but the mixed use of "you" vs. "we" across this page should probably be improved at some point

**Review by kartben — CHANGES_REQUESTED** [review]

**Review by kartben — CHANGES_REQUESTED** [review]
> The go tool documentation is very old, and the tool is not recommended for use

My only request for change is that I'd like to see an entry for this in the release notes to make it extra clear to folks that might still be using this tool that they should probably look for other options. I wouldn't go as far as adding something to the migration guide too, but if you feel there's value in being even more proactive in giving folks recommendations for a replacement, go ahead.

File: samples/subsys/mgmt/mcumgr/smp_svr/README.rst (hunk 1 above)
Comment (nordicjm) [reply samples/subsys/mgmt/mcumgr/smp_svr/README.rst]: Made some changes

**Review by kartben — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Comment by vrahane** [comment]
mcumgr CLI tool has matured quite a bit ever since and go libraries can be updated as needed by devs. We still use it and works great for us, we have updated serial/coap/ble libraries internally. Though, I do understand if you guys are moving away from Go. Not a very widely used language in the embedded world.

**Pull request merged**